### PR TITLE
Increased coverage of owasp/api more than 90%

### LINKED ISSUE #181: Increase backend owasp/api coverage
We don't have any coverage here and want it to be 75%+

```
apps/owasp/api/chapter.py                                                           9      9      0      0     0%   3-28
apps/owasp/api/committee.py                                                         9      9      0      0     0%   3-27
apps/owasp/api/event.py                                                             9      9      0      0     0%   3-28
apps/owasp/api/project.py                                                           9      9      0      0     0%   3-28
apps/owasp/api/search/chapter.py                                                   12     12      2      0     0%   3-39
apps/owasp/api/search/committee.py                                                  8      8      0      0     0%   3-34
apps/owasp/api/search/issue.py                                                     19     19      4      0     0%   3-50
apps/owasp/api/search/project.py                                                   17     17      2      0     0%   3-50
apps/owasp/api/urls.py                                                             10     10      0      0     0%   3-15
```

### PR DESCRIPTION
**Pull request by Naveen-Pal**
Resolves #181 
This PR resolve "Increase backend owasp/api coverage"

In this PR, coverage of the backend owasp/api is increased (which was before 0%). Lead to increase in the total coverage by 7-8%. Below are the snapshots to see the actual coverage of all the file.

Screenshot 2024-12-17 215153
Screenshot 2024-12-17 215213

All the test file are created inside the backend/tests folder and structured the same way as in backend/app folder.
backend/tests/owasp/
├── __init__.py
├── api
│   ├── chapter_tests.py
│   ├── committee_tests.py
│   ├── event_tests.py
│   ├── project_tests.py
│   ├── search
│   │   ├── chapter_search_tests.py
│   │   ├── committee_search_tests.py
│   │   ├── issue_search_tests.py
│   │   └── project_search_tests.py
│   └── urls_tests.py
└── models
    ├── __init__.py
    └── project_tests.py

### CODE REVIEW

File: backend/tests/owasp/api/search/chapter_search_tests.py
Comment (arkid15r) [inline backend/tests/owasp/api/search/chapter_search_tests.py:1]: These file names should not include `_search_`

File: backend/tests/owasp/api/search/committee_search_tests.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+import pytest
+from django.http import HttpRequest, JsonResponse
+
+from apps.owasp.api.search.committee import committees, get_committees
+from apps.owasp.models.committee import Committee
+
+# Test data for parametrized tests
+MOCKED_COORDINATES = (37.7749, -122.4194)
+MOCKED_USER_IP = "127.0.0.1"
```
Comment (arkid15r) [inline backend/tests/owasp/api/search/committee_search_tests.py]: Do you need these outside of Chapter tests?

File: backend/tests/owasp/api/search/committee_search_tests.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+MOCKED_HITS = {
+    ("hits"): [
+        {"idx_name": "OWASP San Francisco", "idx_url": "https://owasp.sf"},
+        {"idx_name": "OWASP New York", "idx_url": "https://owasp.ny"},
+    ],
```
Comment (arkid15r) [inline backend/tests/owasp/api/search/committee_search_tests.py]: The data looks like it's from Chapter tests

File: backend/tests/owasp/api/search/committee_search_tests.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+                "committees": MOCKED_HITS["hits"],
+                "total_pages": MOCKED_HITS["nbPages"],
+            },
+        ),
+    ],
+)
+def test_committees(query, page, expected_response):
+    # Create a mock request
+    request = HttpRequest()
+    request.GET = {"q": query, "page": str(page)}
```
Comment (arkid15r) [inline backend/tests/owasp/api/search/committee_search_tests.py]: Define page as a string in the test data

File: backend/tests/owasp/api/search/committee_search_tests.py [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+        patch(
+            "apps.owasp.api.search.committee.get_committees", return_value=MOCKED_HITS
+        ) as mock_get_committees,
+    ):
+        # Call function
+        response = committees(request)
+
+        # Assertions
+        assert isinstance(response, JsonResponse)
+        assert response.status_code == STATUS_CODE_200
```
Comment (arkid15r) [inline backend/tests/owasp/api/search/committee_search_tests.py]: Use `requests.codes.ok`

**Review by arkid15r — CHANGES_REQUESTED** [review]
The code contains a lot of copy/paste please address the comments I left for each file

**Comment by Naveen-Pal** [comment]
@arkid15r Please check now, I have fixed all issues mentioned.

File: backend/tests/owasp/api/committee_tests.py [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+            },
+            True,
+        ),
+    ],
+)
+def test_committee_serializer_validation(data, expected):
+    serializer = CommitteeSerializer(data=data)
+    is_valid = serializer.is_valid()
+
+    # Assert that the result matches the expected outcome
```
Comment (arkid15r) [inline backend/tests/owasp/api/committee_tests.py]: here

File: backend/tests/owasp/api/search/chapter_tests.py [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+from unittest.mock import patch
+
+import pytest
+import requests
+from django.http import HttpRequest, JsonResponse
+
+from apps.owasp.api.search.chapter import chapters, get_chapters
+from apps.owasp.models.chapter import Chapter
+
+# Test data for parametrized tests
```
Comment (arkid15r) [inline backend/tests/owasp/api/search/chapter_tests.py]: here

File: backend/tests/owasp/api/search/chapter_tests.py [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+    ("nbPages"): 5,
+}
+
+
+@pytest.mark.parametrize(
+    ("query", "page", "expected_hits"),
+    [
+        ("security", "1", MOCKED_HITS),
+        ("web", "2", MOCKED_HITS),
+        ("", "1", MOCKED_HITS),  # Edge case: empty query
```
Comment (arkid15r) [inline backend/tests/owasp/api/search/chapter_tests.py]: maybe not here as it's about the edge case

File: backend/tests/owasp/api/search/chapter_tests.py [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+    ],
+)
+def test_get_chapters(query, page, expected_hits):
+    # Use context managers for mocking
```
Comment (arkid15r) [inline backend/tests/owasp/api/search/chapter_tests.py]: here

File: backend/tests/owasp/api/search/chapter_tests.py [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+    with patch(
+        "apps.owasp.api.search.chapter.raw_search", return_value=expected_hits
+    ) as mock_raw_search:
+        # Call function
+        result = get_chapters(query=query, meta={"REMOTE_ADDR": MOCKED_USER_IP}, page=int(page))
+
+        # Assertions
```
Comment (arkid15r) [inline backend/tests/owasp/api/search/chapter_tests.py]: here

**Review by arkid15r — CHANGES_REQUESTED** [review]
Looks good in general. I'd like to request the excessive comments clean up before the merge (marked some examples as `here`).

We don't need comments when code is easily readable and obvious. Please comment only when it's really needed.

**Comment by Naveen-Pal** [comment]
If you have any other improvement then please share.

**Comment by arkid15r** [comment]
> If you have any other improvement then please share.

No other comments, looks good! Thank you for improving the test coverage -- it's very important for having a stable/reliable application @Naveen-Pal

**Review by arkid15r — APPROVED** [review]
